department-of-veterans-affairs/va.gov-cms
# Vacms 8780 next setup

## Description

See #8780.

* adds drupal/next as a requirement
* adds a couple patches
* enables drupal/next and next-jsonapi and dependencies
* adds a `nextjs_authenticator` role and assigns a few necessary permissions to it
* adds a Nextjs 'site' definition. The function of this is to direct the CMS what URLs to use for previewing.
* adds config associating entities with the Nextjs site, so they can be previewed.

Things outside scope:
* There will be URLs specific to each environment for the Nextjs site definition, and secrets will be pulled from the appropriate locations for those environments. What is committed here is not intended for production. This will be taken care of when setting up preview. https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8785
* `next_entity_type_config` entities should be generated automatically for all bundles of selected types. That is, all node types should have one of these config entities, and if a new node type is created, a config entity should be created for it automatically. https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8786

## QA suggestions
These changes will mostly only provide functionality in the context of the CMS as a provider of content to a Next-Drupal site, which is not yet available in a testable form. As such, QA of this consists of:

1. Ensure that normal functioning of the site continues and does not throw new warnings or errors. Automated testing against the environment should provide a set of 'normal functions', as does running a content-build against this environment. See https://cms-ezfyrhzpzziky2vsiaoem3qw1f4j0gys.ci.cms.va.gov/admin/reports/dblog to examine output.

2. Ensure that the modules, role, site, and entity config that are part of the PR are active. These are:
  * drupal/next and dependencies: https://cms-ezfyrhzpzziky2vsiaoem3qw1f4j0gys.ci.cms.va.gov/admin/modules
  * role/permissions: https://cms-ezfyrhzpzziky2vsiaoem3qw1f4j0gys.ci.cms.va.gov/admin/people/permissions
  * Next site: https://cms-ezfyrhzpzziky2vsiaoem3qw1f4j0gys.ci.cms.va.gov/admin/config/services/next
  * Entity configuration for next-drupal: https://cms-ezfyrhzpzziky2vsiaoem3qw1f4j0gys.ci.cms.va.gov/admin/config/services/next/entity-types
